client/src/lib/fetch.js

```diff
@@ -0,0 +1,18 @@
+import { SERVER_URL } from "../constants";
+
+export const getUserData = async (id = "") => {
+  const getUserDataEndpoint = `${SERVER_URL}/user/${id}`;
+  const result = await fetch(getUserDataEndpoint, {
```

Comment:
nit: rename `result` to `response` or just `res`, since that is the type of the [returned object](https://fetch.spec.whatwg.org/#fetch-method).